src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -1070,7 +1070,7 @@ def _process_stream_response(self, messages, temperature, start_time, formatted_
             tools=formatted_tools,
             start_time=start_time,
             console=self.console,
-            display_fn=display_generating,
+            display_fn=display_generating if (not stream and self.verbose) else None,  # stream is True in this context
```

Ruff (0.12.2)

Undefined name `stream`

(F821)